### ISSUE TITLE
Blynk Legacy closed to new users, cannot compile without Blynk

### ISSUE THREAD

**Issue by rtho782** [jhughes1010/weather#22]
I don't want to use Blynk at all, ideally I'd just use MQTT. I can't sign up for Blynk legacy, only 2.0 which is not compatible anyway.

However, the project won't compile without Blynk credentials even when blynk is not enabled in secrets.h.

I've even commented out blynksimpleesp32.h in the includes, but this then causes loads of further errors.

Is it possible to remove Blynk entirely?

**Comment by jhughes1010** [comment]
It is. This is an older project and I'm not actively making edits. I made
it a point to remove Blynk entirely in version 4 of the project. If I
recall properly, there basic WiFi connect components that were used in the
Blynk library and replacing with a normal WiFi library would be the best
course of action.

**Comment by farzadb** [comment]
I have a version of this code with Blynk removed. I’ll fork and post it shortly.

**Comment by rtho782** [comment]
> I have a version of this code with Blynk removed. I’ll fork and post it shortly.

That would be massively appreciated :)

**Comment by Jimcam65** [comment]
I also would appreciate it :)

**Comment by farzadb** [comment]
I forgot it was already in my repo. You can find it here. Works with the latest Blynk API. https://github.com/farzadb/weather

**Comment by coderpussy** [comment]
@rtho782 @Jimcam65  Hi guys,
is the desire for a version of James' sketch without Blynk and Thingspeak still valid?

I have now completed a slimmed-down version without Blynk and Thingspeak. But with MQTT and local network LAMP stack data storage. It is not yet tested and unfortunately not yet in Git.

If there is interest, I will make this version available on my Github account in the coming days.

**Comment by rtho782** [comment]
Yeah, that would be good! I have my station somewhat working via thingspeak without blynk, but it isn't fully local and if the WiFi disconnects I have to climb on the roof to power cycle it to get it to work again!! :D


________________________________

**Comment by coderpussy** [comment]
Hi @rtho782 for testing purposes, I uploaded the customized sketch to Git in a separate [develop branch](https://github.com/coderpussy/weather/tree/develop). However, it is still untested and certainly needs further adjustments. Sketch was compiled with esp32 library v.3.0.5 and also needed a first customization regarding the WDT.

Please consider it only as a preliminary blueprint and not seriously as a fully executable release version. I will possibly implement further adjustments next weekend. Suggestions and corrections are very welcome.

I have also added some administration files for a local network LAMP stack in the **`admin`** folder from my [v.2.0](https://github.com/coderpussy/SolarTherm-WIFI-Local-Admin) of Debasish's [original project](https://www.instructables.com/Solar-Powered-WiFi-Weather-Station-V20/). However, these are at best a basic idea and also need to be expanded and/or improved.

**Comment by Jimcam65** [comment]
Sweet , greatly appreciated been struggling along each time one of my versions goes down,
Always a great help thank you.
Jim
________________________________

**Comment by coderpussy** [comment]
I just updated my [repository](https://github.com/coderpussy/weather/tree/develop) and fixed a WDT issue, some code structure cosmetics and additonally i added a switch to let the sketch running on both esp32 board library version 2.0.x -> 3.0.x

**Comment by coderpussy** [comment]
@rtho782 @Jimcam65 Hi guys, i added a OTA firmware update functionality based on http request to my [sketch](https://github.com/coderpussy/weather/tree/develop). Currently it does not have a real version check, but a md5 file comparison.

For future dev. the functionality itself should be more bulletproof. In worst case the binary could corrupt during transmission and afterwards the esp could brick or crash or whatever. A first upload to the filesystem should be better.
Also a more secure access to the files should be implemented. But, it's dev. area, right? 😉 

It is working fine in my environment. And as long i don't brick a esp device with this, who cares... 😁

**Comment by Jimcam65** [comment]
Excellent.... I will give this a shot; I think I'm close to getting this to run with your code.
Thanks,
Jim

**Comment by coderpussy** [comment]
@rtho782 @Jimcam65 Hi guys, i added a boot log functionality to my [sketch](https://github.com/coderpussy/weather/tree/develop). Furthermore i added the issues tab on this github repo for easier error reporting or suggestions. So, further discussion can be done there, if someone like.
